src/som/vmobjects/SClass.java

```diff
@@ -56,6 +60,7 @@
   @CompilationFinal private HashSet<SlotDefinition> slots; // includes slots of super classes and mixins
 
   @CompilationFinal private MixinDefinition mixinDef;
+  @CompilationFinal private VirtualFrame context; // a frame containing the outer scope (implicit classes only)
```

Comment:
Kind of the same question as with the builders, why do you need a `context` and `enclosingObject`? Seems, we can only have one of those at the same time?